keps/sig-auth/4447-promote-policyreport-sig-api/README.md

```diff
@@ -0,0 +1,876 @@
+* Cleanup the [existing policy-report directory](https://github.com/kubernetes-sigs/wg-policy-prototypes/tree/master/policy-report) to contain only the necessary content for migration
+* Notify existing projects and tools about the new repository location and share with relevant sigs (`sig-auth`, `sig-security`, `sig-apimachinery`)
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a Kubernetes administrator, I use multiple CNCF projects and tools for security and compliance and would like to have a standard reporting API across these tools.
+
+#### Story 2
+
+As a Kubernetes developer, or a CNCF project developer, I want a standard API to publish reports, so I can leverage common management tools for visibility, governance, and compliance.
+
+#### Story 3
+
+As a Kubernetes management tool developer, I want to use a common API to retrieve policy results, scan results, vulnerability findings, or other reports across my clusters.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+* The Policy Report CRD originated with this proposal back in 2020. Refer to [Policy Report Custom Resource Definition](https://docs.google.com/document/d/1nICYLkYS1RE3gJzuHOfHeAC25QIkFZfgymFjgOzMDVw/edit#heading=h.nhx5d35q5eth).
+* Presented to `sig-auth` about the proposal to make PolicyReport an official SIG API on [June 21, 2023](https://docs.google.com/document/d/1woLGRoONE3EBVx-wTb4pvp4CI7tmLZ6lS26VTbosLKM/view#heading=h.ii52a0istwiv). This KEP is the outcome of that presentation.
+* List of projects using the PolicyReport CRD either from `wg-policy-prototypes` or maintain their own fork. See [this comment](https://github.com/kubernetes-sigs/wg-policy-prototypes/issues/112#issuecomment-1453209660).
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+#### load on etcd
+
+Based on the producer and usage, it is possible to create lots of report objects. 
+For example, if a policy engine has 20 policy rules and a namespace has 1000 pods, 
+an implementation may produce 20,000 reports. This can overwhelm etcd.
```

Comment:
```suggestion
Based on the producer and usage, it is possible to create lots of report objects. 
For example, if a policy engine has 20 policy rules and a namespace has 1000 pods, 
an implementation may produce 20,000 reports. If a cluster operator deploys PolicyReport
into their cluster, using this APU can overwhelm etcd.
```
(is this a risk? We already let people deploy any CRD they like.)

Reply:
It has definitely been reported as an issue for users of the API. Whether that constitutes a risk or not is a good question, but this should be highlighted _somwhere_

Reply:
We have 2 projects we can link to and reference - perhaps in the docs?
- https://github.com/kyverno/reports-server
- https://github.com/open-cluster-management-io/enhancements/tree/main/enhancements/sig-policy/98-long-term-compliance-history

Reply:
This is my core concern with the concept of reporting findings via the API server.

I think events are another example of a high-volume object? One salient difference between this and events, is that events are understood to be subject to throttling/sampling. Security reports may not have the same luxury.

I like the idea of reports-server, but IMO it would need to be an expectation that all clusters have a similar scalable backend solution before reports could be reliably enabled without risking cluster stability.

Reply:
I should also add that there is a difference between "users can deploy any CRD they like" and "K8s accepts using KRM/the API server this way as a valid practice", the second statement has much stronger implications around supportability.

Reply:
@JimBugwadia can you comment on the reliability and performance concerns brought up here?